addons/web/static/src/views/fields/monetary/monetary_field.js

```diff
@@ -64,20 +64,28 @@ export class MonetaryField extends Component {
         return getCurrency(this.currencyId).digits;
     }
 
+    get recordValue(){
+        return this.props.record.data[this.props.name];
+    }
+
     get formattedValue() {
         if (
             this.props.inputType === "number" &&
             !this.props.readonly &&
-            this.props.record.data[this.props.name]
+            this.recordValue
         ) {
-            return this.props.record.data[this.props.name];
+            return this.recordValue;
         }
-        return formatMonetary(this.props.record.data[this.props.name], {
+        return formatMonetary(this.recordValue, {
             digits: this.currencyDigits,
             currencyId: this.currencyId,
             noSymbol: !this.props.readonly || this.props.hideSymbol,
         });
     }
+
+    parseMonetaryInput(text){
```

Comment:
```suggestion
    parse(value) {
```
again, this is how we typically define it (see char_field.js)